src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -326,7 +334,13 @@ public function getAttribute($key)
             return;
         }
 
-        return $this->getRelationValue($key);
+        if ($this->relationLoaded($key) || method_exists($this, $key)) {
```

Comment:
Can we find a better solution for this? We shouldn't have to run the same checks twice.

Can we throw a `PropertyNotFoundException` in `getRelationValue()` and catch it here?

Reply:
I agree it's not ideal, was trying to find a way of avoiding it doing the check twice so any suggestions welcome. Moving the throw to `getRelationValue` wouldn't be an option for a few reasons - 1. that method is public so it changes the behaviour of that method when used in isolation, 2. if your extending & overriding that method again it changes the behaviour, 3. it's a bit quirky to me to throw a `PropertyNotFoundException` in a method that's specifically related to handling getting a relation's value and has nothing to do with attributes on the model.

I would imagine that these checks `relationLoaded` and `method_exists` are pretty quick anyway.

Reply:
> 1. that method is public so it changes the behaviour of that method when used in isolation

I think it would be desirable to have `getRelationValue()` respect `$strictProperties`. After all, the setting is called `$strictProperties`, not `$strictAttributes`.